src/net/udp.rs

```diff
@@ -80,22 +80,35 @@ impl UdpSocket {
         if let Some(link_conditioner) = &self.link_conditioner {
             if link_conditioner.should_send() {
                 for payload in packet_data.parts() {
-                    bytes_sent += self.socket.send_to(&payload, packet.addr()).map_err(|io| {
-                        NetworkError::from(NetworkErrorKind::IOError { inner: io })
-                    })?;
+                    bytes_sent += self.send_packet(&packet.addr(), payload)?;
                 }
             }
         } else {
+            if let Some(dropped) = lock.gather_dropped_packets() {
+                for payload in dropped {
+                    bytes_sent += self.send_packet(&packet.addr(), &payload)?;
+                }
+            }
+
             for payload in packet_data.parts() {
-                bytes_sent += self
-                .socket
-                .send_to(payload, packet.addr())
-                .map_err( |io | NetworkError::from(NetworkErrorKind::IOError { inner: io })) ?;
+                bytes_sent += self.send_packet(&packet.addr(), payload)?;
             }
         }
 
-    Ok(bytes_sent)
-}
+        Ok(bytes_sent)
+    }
+
+    /// Send a single packet over the udp socket.
+    fn send_packet(&self, addr: &SocketAddr, payload: &[u8]) -> NetworkResult<usize>  {
+        let mut bytes_sent = 0;
+
+        bytes_sent += self
+            .socket
+            .send_to(payload, addr)
+            .map_err(|io| NetworkError::from(NetworkErrorKind::IOError { inner: io }))?;
+
+        return Ok(bytes_sent)
```

Comment:
drop the return 😄

Reply:
Clippy lint should be catching these, and if we aren't running it is as part of the CI/CD, we should

Reply:
Alright 😐